libraries/rush-lib/src/logic/operations/ShellOperationRunner.ts

```diff
@@ -150,6 +150,11 @@ export class ShellOperationRunner implements IOperationRunner {
       context.collatedWriter.terminal,
       this._logFilenameIdentifier
     );
+    const cacheProjectLogWritable: ProjectLogWritable = new ProjectLogWritable(
```

Comment:
The act of creating the object opens the log file. This (1) needs to always get cleaned up in a `finally` branch, (2) needs to not prevent the main projectLogWritable from getting cleaned up if it fails to open, and (3) doesn't need to be opened if the build cache feature is completely disabled.

Reply:
Also, the cache log should technically be the outermost log, and also responsible for the skip detection and such.